src/huggingface_hub/hf_api.py

```diff
@@ -3524,7 +3524,7 @@ def delete_repo(
             if not missing_ok:
                 raise
 
-    @validate_hf_hub_args
+    @_deprecate_method(version="0.29", message="Please use `update_repo_settings` instead.")
```

Comment:
```suggestion
    @_deprecate_method(version="0.29", message="Please use `update_repo_settings` instead.")
    @validate_hf_hub_args
```

(still good to validate HF args even if deprecated)